libarchive/archive_write_disk_posix.c

```diff
@@ -4044,10 +4044,10 @@ older(struct stat *st, struct archive_entry *entry)
 {
 	/* First, test the seconds and return if we have a definite answer. */
 	/* Definitely older. */
-	if (st->st_mtime < archive_entry_mtime(entry))
+	if ((time_t) st->st_mtime < archive_entry_mtime(entry))
```

Comment:
This is still broken: if `st->st_mtime` is out of range, then casting will change the meaning here.

See my comments in the bug about an approach that might actually do the right thing.